docs/apps/interchain-accounts/legacy/auth-modules.md

```diff
@@ -0,0 +1,381 @@
+
+The authentication module must:
+
+- Authenticate interchain account owners.
+- Track the associated interchain account address for an owner.
+- Send packets on behalf of an owner (after authentication).
+
+Please note that since ibc-go v6 the channel capability is claimed by the controller submodule and therefore it is not required for authentication modules to claim the capability in the `OnChanOpenInit` callback. When the authentication module sends packets on the channel created for the associated interchain account it can pass a `nil` capability to the legacy function `SendTx` of the controller keeper (see [section `SendTx`](#sendtx) below for mode information). 
+
+## `IBCModule` implementation
+
+The following `IBCModule` callbacks must be implemented with appropriate custom logic:
+
+```go
+// OnChanOpenInit implements the IBCModule interface
+func (im IBCModule) OnChanOpenInit(
+    ctx sdk.Context,
+    order channeltypes.Order,
+    connectionHops []string,
+    portID string,
+    channelID string,
+    chanCap *capabilitytypes.Capability,
+    counterparty channeltypes.Counterparty,
+    version string,
+) (string, error) {
+    // since ibc-go v6 the authentication module *must not* claim the channel capability on OnChanOpenInit
+
+    // perform custom logic
+
+    return version, nil
+}
+
+// OnChanOpenAck implements the IBCModule interface
+func (im IBCModule) OnChanOpenAck(
+    ctx sdk.Context,
+    portID,
+    channelID string,
+    counterpartyVersion string,
+) error {
+    // perform custom logic
+
+    return nil
+}
+
+// OnChanCloseConfirm implements the IBCModule interface
+func (im IBCModule) OnChanCloseConfirm(
+    ctx sdk.Context,
+    portID,
+    channelID string,
+) error {
+    // perform custom logic
+
+    return nil
+}
+
+// OnAcknowledgementPacket implements the IBCModule interface
+func (im IBCModule) OnAcknowledgementPacket(
+    ctx sdk.Context,
+    packet channeltypes.Packet,
+    acknowledgement []byte,
+    relayer sdk.AccAddress,
+) error {
+    // perform custom logic
+
+    return nil
+}
+
+// OnTimeoutPacket implements the IBCModule interface.
+func (im IBCModule) OnTimeoutPacket(
+    ctx sdk.Context,
+    packet channeltypes.Packet,
+    relayer sdk.AccAddress,
+) error {
+    // perform custom logic
+
+    return nil
+}
+```
+
+The following functions must be defined to fulfill the `IBCModule` interface, but they will never be called by the controller submodule so they may error or panic.
```

Comment:
I think it's always useful to remind people that ICA is asymmetrical, which explains why the below call backs should panic